canonical-web-and-design/maas-ui
# refactor(shared): make shared click-outside hook

## Done

- Move the click-outside code to a shared hook.
- Update shared workspace to using consistent linting.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the config tab in machine details.
- Click on the tag form Edit button.
- Click on the tag input and the dropdown should appear.
- Click outside the dropdown and it should close.

## Fixes

Fixes: canonical-web-and-design/app-tribe#835.
